source/common/quic/quic_filter_manager_connection_impl.cc

```diff
@@ -87,7 +87,6 @@ void QuicFilterManagerConnectionImpl::close(Network::ConnectionCloseType type) {
   } else if (hasDataToWrite()) {
     // Quic connection has unsent data but caller wants to close right away.
     ASSERT(type == Network::ConnectionCloseType::NoFlush);
-    quic_connection_->OnCanWrite();
```

Comment:
Why omitting OnCanWrite() here?

Reply:
OnCanWrite is causing a QUIC_BUG to fire because it's making us write while we're in the middle of packet processing. Removing this solves that issue, and isn't causing any test regressions.